ucp/benchmarks/backends/tornado.py

```diff
@@ -0,0 +1,121 @@
+import asyncio
+from time import monotonic
+
+from tornado.iostream import StreamClosedError
+from tornado.tcpclient import TCPClient
+from tornado.tcpserver import TCPServer
+
+from ucp.benchmarks.backends.base import BaseClient, BaseServer
+
+
+class TornadoServer(BaseServer):
+    def __init__(self, args, xp, queue):
+        self.args = args
+        self.xp = xp
+        self.queue = queue
+
+    def has_cuda_support() -> bool:
+        return False
+
+    def _start_listener(self, server, port):
+        if port is not None:
+            server.listen(port)
+        else:
+            for i in range(10000, 60000):
+                try:
+                    server.listen(i)
+                except OSError:
+                    continue
+                else:
+                    port = i
+                    break
```

Comment:
You can `server.listen(0)` to get a random port, but then it appears very difficult to actually determine what the port you got was, so meh I guess.